src/common/schedulers/slurm_commands.py

```diff
@@ -165,6 +165,14 @@ def _batch_node_info(nodenames, nodeaddrs, nodehostnames, batch_size):
         # Only split on , if there is ] before
         # For ex. "node-[1,3,4-5],node-[20,30]" should split into ["node-[1,3,4-5]","node-[20,30]"]
         nodenames = re.split("(?<=]),", nodenames)
+    if nodeaddrs and nodehostnames:
```

Comment:
The input of nodeaddrs here can be a string

Reply:
I refactored the code